### PR TITLE
store: do not unnecessarily catch/rewrite exceptions

### PR DESCRIPTION
**Pull request by cjp256**
The exceptions may have information that is necessary to
debug:

- _try_login() unnecessarily caught a few exceptions, calling
  failed_login() to report them.  Instead, just surface the
  applicable exceptions (which have similar error messages).

- Similarily, drop _requires_login() wrapper which also
  unnecessarily caught storeapi.errors.InvalidCredentialsError and
  raised an identical error message.

- Update tests, dropping a few duplicate test cases.

- Remove unneeded AUTHENTICATION_ERROR, and now-unneeded
  INVALID_CREDENTIALS and ACCOUNT_INFORMATION_ERROR constants.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Review by sergiusens — APPROVED** [review]

**Review by sergiusens — CHANGES_REQUESTED** [review]
There are leftover `if snapcraft.login` blocks in snapcraft.cli.store

File: snapcraft/_store.py [hunk 1]
```diff
@@ -547,11 +547,8 @@ def _maybe_prompt_for_key(name):
 def register_key(name):
     key = _maybe_prompt_for_key(name)
     store_client = StoreClientCLI()
-    try:
-        if not login(store=store_client, acls=["modify_account_key"], save=False):
-            raise storeapi.errors.LoginRequiredError()
-    except storeapi.errors.StoreAuthenticationError as e:
-        raise storeapi.errors.LoginRequiredError(str(e)) from e
```
Comment (sergiusens) [inline snapcraft/_store.py:562]: is LoginRequiredError still needed or can it be removed?

Comment (cjp256) [reply snapcraft/_store.py:562]: Removed! thnx

**Review by cmatsuoka — APPROVED** [review]
+1. Internal handling of these exceptions looks unnecessary, just sending them upwards is better (and simpler).

**Pull request merged**